File: Android/ProjectoESeminario/app/src/main/java/andre/pt/projectoeseminario/State/ClipboardControllerFactory.java
package andre.pt.projectoeseminario.State;


/**
 * Factory so we return always the same object of the ClipboardController class
 */
public class ClipboardControllerFactory {
    private static ClipboardController clipboardController;

    public static ClipboardController getSingleton(String data){
        if(clipboardController == null)
            clipboardController = new ClipboardController(data);

        return clipboardController;
    }

}

File: Windows/ProjectoESeminario/ProjectoESeminario/Controller/State/ClipboardControllerFactory.cs
using System;

namespace ProjectoESeminario.Controller.State
{
    public class ClipboardControllerFactory
    {
        private static ClipboardController clipboardController;


        public static ClipboardController getSingleton(long order)
        {
            return clipboardController ?? (clipboardController = new ClipboardController(order));
        }

    }
}
